include/beast/http/basic_parser_v1.hpp

```diff
@@ -412,6 +431,24 @@ class basic_parser_v1
     needs_eof(std::false_type) const;
 
     template<class C>
+    class has_on_start_t
+    {
+        template<class T, class R =
+            decltype(std::declval<T>().on_start(
+                std::declval<error_code&>()),
+                    std::true_type{})>
+        static R check(int);
+        template <class>
+        static std::false_type check(...);
+        using type = decltype(check<C>(0));
+    public:
+        static bool const value = type::value;
+    };
+    template<class C>
+    using has_on_start =
```

Comment:
Yeah unfortunately I'm limited to C++11 here so we can't do the `_v` or the modern `_t`.